sopel/modules/reddit.py

```diff
@@ -87,7 +87,11 @@ def get_is_cakeday(entrytime):
 @plugin.url(image_url)
 @plugin.output_prefix(PLUGIN_OUTPUT_PREFIX)
 def image_info(bot, trigger, match):
-    url = match.group(0)
+    preview = match.group("subdomain") == "preview"
+    if preview:
+        url = "https://i.redd.it/{}".format(match.group("image"))
+    else:
+        url = match.group(0)
```

Comment:
I think @Exirel would agree that it would be better to leave the original `url = match.group(0)` assignment, and only add an `if preview:` block. We try to avoid having `else:` in cases like this, just for style reasons.

(Which is to say, we try to avoid _adding_ an `else:` in cases like this, because preexisting code certainly has plenty of them.)

Reply:
Regarding what you said on IRC about avoiding the function call: The performance impact is meaningless in practice (these tests used `timeit`'s default of 1,000,000 iterations), but not zero.

<details>
<summary>Test script</summary>

```python
import timeit

no_else_i = timeit.timeit('''
url = m.group(0)
if preview:
    url = 'https://i.redd.it/{}'.format(m.group('image'))
''',
r'''
import re
r = re.compile(r'https?://(?P<subdomain>i|preview)\.redd\.it/(?P<image>[^?\s]+)')
m = r.match('https://i.redd.it/afheihwoi61.png')
preview = m.group('subdomain') == 'preview'
''')

no_else_preview = timeit.timeit('''
url = m.group(0)
if preview:
    url = 'https://i.redd.it/{}'.format(m.group('image'))
''',
r'''
import re
r = re.compile(r'https?://(?P<subdomain>i|preview)\.redd\.it/(?P<image>[^?\s]+)')
m = r.match('https://preview.redd.it/afheihwoi61.png?some=get&params=here')
preview = m.group('subdomain') == 'preview'
''')

yes_else_i = timeit.timeit('''
if preview:
    url = 'https://i.redd.it/{}'.format(m.group('image'))
else:
    url = m.group(0)
''',
r'''
import re
r = re.compile(r'https?://(?P<subdomain>i|preview)\.redd\.it/(?P<image>[^?\s]+)')
m = r.match('https://i.redd.it/afheihwoi61.png')
preview = m.group('subdomain') == 'preview'
''')

yes_else_preview = timeit.timeit('''
if preview:
    url = 'https://i.redd.it/{}'.format(m.group('image'))
else:
    url = m.group(0)
''',
r'''
import re
r = re.compile(r'https?://(?P<subdomain>i|preview)\.redd\.it/(?P<image>[^?\s]+)')
m = r.match('https://preview.redd.it/afheihwoi61.png?some=get&params=here')
preview = m.group('subdomain') == 'preview'
''')

print('if/else:')
print('  i.       {}'.format(yes_else_i))
print('  preview. {}'.format(yes_else_preview))
print('if only:')
print('  i.       {}'.format(no_else_i))
print('  preview. {}'.format(no_else_preview))
```

</details>

Result:

```
if/else:
  i.       0.04802380000000994
  preview. 0.17630289999988236
if only:
  i.       0.0444231999999829
  preview. 0.23507989999961865
```

Reply:
Additionally, `preview` is the uncommon case, so eh. I'll go for tidy.

Reply:
(I agree with @dgw )